### PR TITLE
Allow volunteers to mark themselves as unavailable

### PR DESCRIPTION
**Pull request by joelmichael**
Implementation for this feature request: https://github.com/OfficeMomsandDads/scheduler/issues/212

The need show page has some new functionality. There is a "Responses" section beneath the shifts. There, notified users will be able to mark themselves as unavailable for this need. Also, admins, coordinators and social workers will be able to see who has marked themselves unavailable, as well as all notified volunteers who have not yet claimed a shift nor marked themselves unavailable.

If marked unavailable, a user will not receive any more notifications related to this need, *unless* a new shift is created. If a new shift is created, the unavailable users list is wiped clean, and users will have to mark themselves as unavailable again (or claim the new shift).

Includes some simple JS to toggle the lists of users.

### CODE REVIEW

**Comment by joelmichael** [comment]
I added some tests for the need show view as assert_selects in needs_controller_spec (I was going off https://guides.rubyonrails.org/testing.html). I'm considering doing this in a more rspec-style with a view spec however. Also, I think I want the "Response" h2 to not show up at all unless at least one user is in notified_user_ids.

**Comment by joelmichael** [comment]
@benjaminwood I made some changes to address your comments. I also decided to remove the big "Response" h2 tag since it's not really necessary (we are still in the context of shifts). Users who claim a shift *after* marking themselves unavailable will also now be removed from the unavailable users list. Let me know what you think.

File: app/policies/need_policy.rb [hunk 1]
```diff
@@ -29,6 +29,14 @@ def destroy?
     create?
   end
 
+  def mark_unavailable?
```
Comment (dinjas) [inline app/policies/need_policy.rb]: these could be aliases. 

eg. 
`alias mark_unavailable? show?`

Comment (joelmichael) [reply app/policies/need_policy.rb]: I suppose, it's not how the rest of them are done though

Comment (dinjas) [reply app/policies/need_policy.rb]: true

**Pull request merged**